platform/security/rest/security-rest-clientapi/src/main/java/org/codice/ddf/cxf/client/ClientFactoryFactory.java

```diff
@@ -222,6 +223,32 @@
       String clientSecret,
       String oauthFlow);
 
+  /**
+   * Returns an initialized SecureCxfClientFactory
+   *
+   * @param endpointUrl
+   * @param interfaceClass
+   * @param sourceId
+   * @param discoveryUrl
+   * @param clientId
+   * @param clientSecret
+   * @param username
+   * @param password
+   * @param additionalParameters
+   * @param <T>
+   * @return
+   */
+  <T> SecureCxfClientFactory<T> getSecureCxfClientFactory(
+      String endpointUrl,
+      Class<T> interfaceClass,
+      String sourceId,
+      String discoveryUrl,
+      String clientId,
+      String clientSecret,
+      String username,
+      String password,
+      Map<String, String> additionalParameters);
```

Comment:
✏️ ❓ Should this be called `additionalOauthParameters` instead? It is named that way in `SecurityCxfClientFactoryImpl`.